qualang_tools/octave_tools/calibration_result_plotter.py

```diff
@@ -0,0 +1,739 @@
+import logging
+from typing import Optional
+
+import matplotlib.pyplot as plt
+import numpy as np
+from matplotlib.colors import LinearSegmentedColormap
+from matplotlib.patches import Rectangle
+from qm.octave.octave_mixer_calibration import MixerCalibrationResults
+from qm.type_hinting.general import Number
+from qualang_tools.units import unit
+
+logger = logging.getLogger(__name__)
+
+colors = ["black", "midnightblue", "navy", "darkblue", "mediumblue", "dodgerblue", "white", "lightgrey"]
+
+custom_cmap = LinearSegmentedColormap.from_list("custom_diverging", colors, N=256)
+
+u = unit(coerce_to_integer=True)
+
+
+def show_lo_result(
+    data: MixerCalibrationResults,
+    lo_freq: Optional[Number] = None,
+    label: str = "",
+) -> None:
+    """
+    Show the result of LO leakage automatic calibration.
+
+    Args:
+        data (MixerCalibrationResults): A dictionary containing the calibration results.
+        lo_freq (Optional[float]): The LO frequency in Hz. If not provided, the first LO frequency in data is used.
+        label (str): A label to be used in the plot title.
+    """
+    if lo_freq is None:
+        if len(data) == 1:
+            lo_freq = list(data.keys())[0][0]
+
+        else:
+            print(f"There is a calibration data for {len(data)} LO frequencies. You must choose one")
+            return
+
+    try:
+        output_gain = list(data.keys())[0][1]
```

Comment:
This looks risky to me, because of the non controlled ordering of the dictionary keys, why not simply accessing it with the correct key?